File: src/main/java/com/eager/questioncloud/question/repository/QuestionJpaRepository.java
package com.eager.questioncloud.question.repository;

import com.eager.questioncloud.question.entity.QuestionEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface QuestionJpaRepository extends JpaRepository<QuestionEntity, Long> {
    Optional<QuestionEntity> findByIdAndCreatorId(Long id, Long creatorId);
}

File: qc-question/src/main/kotlin/com/eager/questioncloud/question/infrastructure/repository/QuestionJpaRepository.kt
package com.eager.questioncloud.question.infrastructure.repository

import com.eager.questioncloud.question.infrastructure.entity.QuestionEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface QuestionJpaRepository : JpaRepository<QuestionEntity, Long> {
    fun findByIdAndCreatorId(id: Long, creatorId: Long): Optional<QuestionEntity>
}
